www/src/pages/headless/combobox.md

```diff
@@ -0,0 +1,432 @@
+Clicking on an item when the list is open selects it and closes the list.
+
+## Keyboard Interaction
+
+| Command                                       | Description                    |
+|:----------------------------------------------|:-------------------------------|
+| [[⬆]] [[⬇]] when listbox is open              | Activates previous / next item |
+| [[Home]] [[End]] when listbox is open         | Activates first / last item    |
+| [[Esc]] when listbox is open                  | Closes the combobox            |
+| [[Enter]] [[Space]] when the combobox is open | Selects the active item        |
+
+## Performance
+
+The combo box component used a sepcific internal pipeline to filter and display the selection items:
+
+1) Retrieve the user's input
+2) _Debounce_
+3) Find matches
+4) _Debounce_
+5) Render matches
+
+The debouncing is in place because the above worklfow consists of two relatively expensive operations: _filtering_ and
+_rendering_.
+
+While typing, the query may be manipulated multiple times per second. In order for the filter function to run as few
+times as possible, the flow of inputs is debounced.
+
+The same goes for the actual rendering: It is by far the most expensive operation in the workflow so it is debounced to
+not be executed multiple times in a row.
+
+Adding to the above, the number of displayed items in the dropdown also has an impact on the rendering performance.
+
+::: info
+Most of the time, the default behavior should be working fine. There might be cases, however, where the implementing
+component has a consistently high/low amount of items or other niche scenarios. In those cases, the debouncing and
+other performance-related parameters can be configured via the DSL.
+:::
+
+| Property                | Type   | Default | Description                                                      |
+|:------------------------|:-------|:--------|:-----------------------------------------------------------------|
+| `maximumDisplayedItems` | `Int`  | `20`    | Maximum number of items to display                               |
+| `inputDebounceMillis`   | `Long` | `50L`   | Time to wait and debounce before the filter function is invoked  |
+| `renderDebounceMillis`  | `Long` | `50L`   | Time to wait and debounce before the filter results are rendered |
+
+See [`combobox`](#combobox) for more api information.
+
+## API
+
+### Summary / Sketch
+
+```kotlin
+combobox<T> {
+    val items: ItemsHook()
+    // params: List<T> / Flow<List<T>>
+
+    var itemFormat: (T) -> String
+
+    val value: DatabindingProperty<T>
+
+    var filterBy: FilterFunctionProperty
+    // params: (Sequence<T>, String) -> Sequence<T> / T.() -> String
+
+    val selectionStrategy: SelectionStrategyProperty
+    // methods: autoSelectMatches() / manual()
+
+    var maximumDisplayedItems: Int = 20
+    var inputDebounceMillis: Long = 50L
+    var renderDebounceMillis: Long = 50L
+
+    comboboxPanelReference() { }
```

Comment:
I would suggest to add explicitely an uncommented code block within that shows, that you often (if not almost always) put the `comboboxInput` brick somwhere *indise* this refenrece block.

Does the same apply to `label` or other bricks? If so, at least present them comma separated to make this possible and intended structure explicit.

Reply:
I just saw that you changed this in the API sketch of the KDoc of the `combobox`-factory functions.

I propose to diff both again to see missing / changed things ;-)

```
 *     comboboxPanelReference() {
 *         // this brick is often used with a nested
 *         // comboboxInput() { }
 *     }
 ```